Sources/MapboxCoreNavigation/PassiveLocationDataSource.swift

```diff
@@ -27,6 +27,10 @@ open class PassiveLocationDataSource: NSObject {
         
         self.systemLocationManager.delegate = self
     }
+
+    deinit {
+        try! self.navigator.setElectronicHorizonObserverFor(nil)
```

Comment:
Is it correct to reset EH observer here (and in `RouteController.deinit`)? The only place where it is set up - is `CoreNavigationNavigator` private initializer which is called just once, when instantiating a shared `Navigator`. Thus, running a second active guidance session will result in having no EHObserver set.

Reply:
This is a good point. Seems like we should install the observer for the free drive case here in the PassiveLocationDataSource in `startUpdatingLocation()` and in the `RouteController()` so a delegate is set per drive session, active or passive.